view/src/components/todo.js

```diff
@@ -2,25 +2,422 @@ import React, { Component } from "react";
+import Grid from "@material-ui/core/Grid";
+import Card from "@material-ui/core/Card";
+import CardActions from "@material-ui/core/CardActions";
+import CircularProgress from "@material-ui/core/CircularProgress";
+import CardContent from "@material-ui/core/CardContent";
+import MuiDialogTitle from "@material-ui/core/DialogTitle";
+import MuiDialogContent from "@material-ui/core/DialogContent";
+
+import axios from "axios";
+import dayjs from "dayjs";
+import relativeTime from "dayjs/plugin/relativeTime";
+import { authMiddleWare } from "../util/auth";
 
 const styles = (theme) => ({
   content: {
     flexGrow: 1,
     padding: theme.spacing(3),
   },
+  appBar: {
+    position: "relative",
+  },
+  title: {
+    marginLeft: theme.spacing(2),
+    flex: 1,
+  },
+  submitButton: {
+    display: "block",
+    color: "white",
+    textAlign: "center",
+    position: "absolute",
+    top: 14,
+    right: 10,
+  },
+  floatingButton: {
+    position: "fixed",
+    bottom: 0,
+    right: 0,
+  },
+  form: {
+    width: "98%",
+    marginLeft: 13,
+    marginTop: theme.spacing(3),
+  },
   toolbar: theme.mixins.toolbar,
+  root: {
+    minWidth: 470,
+  },
+  bullet: {
+    display: "inline-block",
+    margin: "0 2px",
+    transform: "scale(0.8)",
+  },
+  pos: {
+    marginBottom: 12,
+  },
+  uiProgess: {
+    position: "fixed",
+    zIndex: "1000",
+    height: "31px",
+    width: "31px",
+    left: "50%",
+    top: "35%",
+  },
+  dialogeStyle: {
+    maxWidth: "50%",
+  },
+  viewRoot: {
+    margin: 0,
+    padding: theme.spacing(2),
+  },
+  closeButton: {
+    position: "absolute",
+    right: theme.spacing(1),
+    top: theme.spacing(1),
+    color: theme.palette.grey[500],
+  },
+});
+
+const Transition = React.forwardRef(function Transition(props, ref) {
+  return <Slide direction="up" ref={ref} {...props} />;
 });
 
-class Todo extends Component {
+class todo extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      todos: "",
+      title: "",
+      body: "",
+      todoId: "",
+      errors: [],
+      open: false,
+      uiLoading: true,
+      buttonType: "",
+      viewOpen: false,
+    };
+
+    this.deleteTodoHandler = this.deleteTodoHandler.bind(this);
+    this.handleEditClickOpen = this.handleEditClickOpen.bind(this);
+    this.handleViewOpen = this.handleViewOpen.bind(this);
+  }
+
+  handleChange = (event) => {
+    this.setState({
+      [event.target.name]: event.target.value,
+    });
+  };
+
+  componentWillMount = () => {
+    authMiddleWare(this.props.history);
+    const authToken = localStorage.getItem("AuthToken");
+    axios.defaults.headers.common = { Authorization: `${authToken}` };
+    axios
+      .get("/todos")
+      .then((response) => {
+        this.setState({
+          todos: response.data,
+          uiLoading: false,
+        });
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  };
+
+  deleteTodoHandler(data) {
```

Comment:
I know these functions are currently fairly self-documenting, but some JSDoc comments would go a long way here, particularly to document the type and nature of the `data` parameter.  Could you add some brief documentation to each of the event handler functions?